contracts/vaults/YearnVaults/base/RibbonVault.sol

```diff
@@ -0,0 +1,744 @@
+import {VaultLifecycleYearn} from "../../../libraries/VaultLifecycleYearn.sol";
+import {ShareMath} from "../../../libraries/ShareMath.sol";
+import {DSMath} from "../../../vendor/DSMath.sol";
+import {IOtoken} from "../../../interfaces/GammaInterface.sol";
+import {IWETH} from "../../../interfaces/IWETH.sol";
+import {IGnosisAuction} from "../../../interfaces/IGnosisAuction.sol";
+import {
+    IStrikeSelection,
+    IOptionsPremiumPricer
+} from "../../../interfaces/IRibbon.sol";
+
+contract RibbonVault is OptionsVaultYearnStorage, DSMath {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using ShareMath for Vault.DepositReceipt;
+
+    /************************************************
+     *  IMMUTABLES & CONSTANTS
+     ***********************************************/
+
+    address public immutable WETH;
+    address public immutable USDC;
+
+    uint256 public constant delay = 1 hours;
+
+    uint256 public constant period = 7 days;
+
+    uint256 public constant YEARN_WITHDRAWAL_BUFFER = 5; // 0.05%
+
+    uint256 public constant YEARN_WITHDRAWAL_SLIPPAGE = 5; // 0.05%
+
+    uint128 internal constant PLACEHOLDER_UINT = 1;
+
+    // GAMMA_CONTROLLER is the top-level contract in Gamma protocol
+    // which allows users to perform multiple actions on their vaults
+    // and positions https://github.com/opynfinance/GammaProtocol/blob/master/contracts/Controller.sol
+    address public immutable GAMMA_CONTROLLER;
+
+    // MARGIN_POOL is Gamma protocol's collateral pool.
+    // Needed to approve collateral.safeTransferFrom for minting otokens.
+    // https://github.com/opynfinance/GammaProtocol/blob/master/contracts/MarginPool.sol
+    address public immutable MARGIN_POOL;
+
+    // GNOSIS_EASY_AUCTION is Gnosis protocol's contract for initiating auctions and placing bids
+    // https://github.com/gnosis/ido-contracts/blob/main/contracts/EasyAuction.sol
+    address public immutable GNOSIS_EASY_AUCTION;
+
+    // Yearn registry contract
+    address public immutable YEARN_REGISTRY;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    event Deposit(address indexed account, uint256 amount, uint16 round);
+
+    event Redeem(address indexed account, uint256 share, uint16 round);
+
+    event WithdrawalFeeSet(uint256 oldFee, uint256 newFee);
+
+    event ManagementFeeSet(uint256 managementFee, uint256 newManagementFee);
+
+    event PerformanceFeeSet(uint256 performanceFee, uint256 newPerformanceFee);
+
+    event CapSet(uint256 oldCap, uint256 newCap, address manager);
+
+    event Withdraw(
+        address account,
+        bool keepWrapped,
+        uint256 amount,
+        uint256 shares
+    );
+
+    event InitiateWithdraw(address account, uint256 shares, uint16 round);
+
+    event CollectVaultFees(
+        uint256 performanceFee,
+        uint256 vaultFee,
+        uint256 round
```

Comment:
I must have missed this before, we should replace the `vaultFee` for `managementFee`

Reply:
I actually added managementFee as well. 

vaultFee is how much we actually made that week which is worth passing in

Reply:
The dev can just do `performanceFee + managementFee = vaultFee` so it's redundant?